File: Machine Learning A-Z Template Folder/Part two - Regression/Section 6 - Polynomial Regression/poly_reg.r
#Data preprocess template

#Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]


# Split the dataset into the Training set and test set
# library(caTools)
# set.seed(123)
# split = sample.split(dataset$Purchased, SplitRatio = 0.8)
# training_set = subset(dataset, split==TRUE)
# test_set = subset(dataset, split==FALSE)


# Featured Scaling
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])

# Fitting Linear Regression to the dataset
lin_reg = lm(formula = Salary ~ ., data = dataset)


# Fitting Polynomial Regression to Dataset
dataset$Level2 = dataset$Level^2
dataset$Level3 = dataset$Level^3
dataset$Level4 = dataset$Level^4
dataset$Level6 = dataset$Level^6

poly_reg = lm(formula = Salary ~ ., data = dataset)


#Visualizing Linear Regression Results
ggplot() + 
  geom_point(aes(x = dataset$Level, y = dataset$Salary), 
             color = 'red') +
  geom_line(aes(x=dataset$Level, y = predict(lin_reg, newdata = dataset)),
            color = 'blue') +
  ggtitle('Truth Or Bluff') +
  xlab("Level") +
  ylab("Salary")


# Visualizing Poly Regression To Dataset
ggplot() + 
  geom_point(aes(x = dataset$Level, y = dataset$Salary), 
             color = 'red') +
  geom_line(aes(x=dataset$Level, y = predict(poly_reg, newdata = dataset)),
            color = 'blue') +
  ggtitle('Truth Or Bluff (Poly)') +
  xlab("Level") +
  ylab("Salary")


# Predicting a new result with linear Regression
y_pred = predict(lin_reg, data.frame(Level = 6.5))

# Predicting a new result with Poly Regression
y_pred = predict(poly_reg, data.frame(Level = 6.5, 
                                      Level2 = 6.5^2,
                                      Level3 = 6.5^3,
                                      Level4 = 6.5^4,
                                      Level6 = 6.5^6))

File: Machine Learning A-Z Template Folder/Part two - Regression/Section 6 - Polynomial Regression/poly_reg.py
# Ploynomial Regression


# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

'''
# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)'''

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""

# Fitting Linear Regression to the dataset
from sklearn.linear_model import LinearRegression
lin_reg = LinearRegression()
lin_reg.fit(X, y)

# Fitting Polynomial Regression to the Dataset
from sklearn.preprocessing import PolynomialFeatures
poly_reg = PolynomialFeatures(degree = 6)
X_poly = poly_reg.fit_transform(X)
#print(X_poly)

lin_reg_2 = LinearRegression()
lin_reg_2.fit(X_poly, y)

# Visualizing linear Regression Results
plt.scatter(X, y, color = "red")
plt.plot(X, lin_reg.predict(X), color = "blue")
plt.title('Truth or Bluff (Linear Regression)')
plt.xlabel("Position Level")
plt.ylabel('Salary')
#plt.show()
 

# Visualizing Polynomial Regression Results
x_grid = np.arange(min(X), max(X), 0.1)
x_grid = x_grid.reshape(len(x_grid), 1)
plt.scatter(X, y, color = "red")
plt.plot(x_grid, lin_reg_2.predict(poly_reg.fit_transform(x_grid)), color = "blue")
plt.title('Truth or Bluff (Polynomial Regression)')
plt.xlabel("Position Level")
plt.ylabel('Salary')
#plt.show()

# Predicting new Result with Linear Regression
#print(lin_reg.predict(6.5))

# Predicitng a new result with Polynomial Regression
print(lin_reg_2.predict(poly_reg.fit_transform(6.5)))